src/api/v4/dataview/histogram.js

```diff
@@ -0,0 +1,186 @@
+  this._bins = this._options.bins;
+}
+
+Histogram.prototype = Object.create(Base.prototype);
+
+Histogram.prototype.DEFAULTS = {
+  bins: 10
+};
+
+/**
+ * Return the resulting data
+ *
+ * @return {HistogramData}
+ * @api
+ */
+Histogram.prototype.getData = function () {
+  if (this._internalModel) {
+    return this._parseData(this._internalModel.get('data'), this._internalModel.get('nulls'), this._internalModel.get('totalAmount'));
+  }
+  return null;
+};
+
+/**
+ * Return the totals data (not affected by filters)
+ *
+ * @return {HistogramData}
+ * @api
+ */
+
+Histogram.prototype.getTotalsData = function () {
+  if (this._internalModel && this._internalModel.getUnfilteredData()) {
+    var totals = this._internalModel.getUnfilteredDataModel();
+    return this._parseData(this._internalModel.getUnfilteredData(), totals.get('nulls'), totals.get('totalAmount'));
+  }
+  return null;
+};
+
+/**
+ * Set number of bins
+ * 
+ * @param {number} bins
+ * @return {carto.dataview.Histogram} this
+ * @api
+ */
+Histogram.prototype.setBins = function (bins) {
+  this._validateBins(bins);
+  var floorBins = Math.floor(bins);
+  this._changeProperty('bins', floorBins);
+  return this;
+};
+
+/**
+ * Return the current number of bins
+ * 
+ * @return {number} Current number of bins
+ * @api 
+ */
+Histogram.prototype.getBins = function () {
+  return this._bins;
+};
+
+/**
+ * Return the distribution type of the current data according to Galtung’s AJUS System http://druedin.com/2012/12/08/galtungs-ajus-system/
+ * 
+ * @return {string} Distribution type of current data
+ * @api 
+ */
+Histogram.prototype.getDistributionType = function () {
+  if (this._internalModel) {
+    var data = this._internalModel.getData();
+    return this._internalModel.getDistributionType(data);
+  }
+  return null;
+};
+
+/**
+ * Return the distribution type of the totals data according to Galtung’s AJUS System http://druedin.com/2012/12/08/galtungs-ajus-system/
+ * 
+ * @return {string} Distribution type of current data
+ * @api 
+ */
+Histogram.prototype.getTotalsDistributionType = function () {
+  if (this._internalModel) {
+    var data = this._internalModel.getUnfilteredData();
+    return this._internalModel.getDistributionType(data);
+  }
+  return null;
+};
+
+Histogram.prototype._validateBins = function (bins) {
+  if (!_.isFinite(bins) || bins < 1) {
+    throw new TypeError('Bins must be a positive value.');
+  }
+};
+
+Histogram.prototype._parseData = function (data, nulls, totalAmount) {
+  if (data) {
+    /**
+     * @typedef {object} BinItem
+     * @property {number} index
+     * @property {number} min - Only appears if freq > 0
+     * @property {number} max - Only appears if freq > 0
+     * @property {number} avg - Only appears if freq > 0
+     * @property {number} freq
+     * @property {number} normalized
+     * @api
+     */
+    /**
+     * @typedef {object} HistogramData
+     * @property {number} nulls
```

Comment:
We should explain these properties a bit better (including descriptions).

Reply:
Same.